src/metabase/db.clj

```diff
@@ -203,15 +203,13 @@
    Returns true if update modified rows, false otherwise."
   [entity entity-id & {:as kwargs}]
   {:pre [(integer? entity-id)]}
-  (let [obj (->> (assoc kwargs :id entity-id)
-                 (models/pre-update entity)
-                 (models/internal-pre-update entity)
-                 (#(dissoc % :id)))
-        result (-> (k/update entity (k/set-fields obj) (k/where {:id entity-id}))
-                   (> 0))]
-    (when result
-      (models/post-update entity (assoc obj :id entity-id)))
-    result))
+  (let [obj           (models/do-pre-update entity (assoc kwargs :id entity-id))
+        rows-affected (k/update entity
+                                (k/set-fields (dissoc obj :id))
+                                (k/where {:id entity-id}))]
+    (when (> rows-affected 0)
+      (models/post-update obj))
+    obj))
```

Comment:
it looks like you are changing the method signature of this method and returning the object now instead of the row count.  i'm cool with that as it probably is more useful this way, but we should make sure this doesn't have any unexpected consequences elsewhere in the code.  and we should update the docstring to match as well.